### ISSUE TITLE
how to use a anonymous plugin for create by myself

### ISSUE THREAD

**Issue by lichnow** [mozilla-neutrino/webpack-chain#75]
```
config.plugin('then')
            .use({
                apply: (compiler) => {
                    compiler.hooks.afterEmit.tap('AfterEmitPlugin', (compilation) => {
                        console.log('worked!')
                    })
                }
            })
```
the error
```
TypeError: Cannot read property '__expression' of undefined
```

but webpack4 support to create anonymous plugin,and in webpack plugins config array,it works fine,how to run in chain?

**Comment by edmorley** [comment]
@yyx990803 I don't suppose you have any thoughts about this? :-)

**Comment by endiliey** [comment]
You can do something like

```js
// myPlugin.js
class MyPlugin {
  apply(compiler) {
      compiler.hooks.compile.tap('WebpackNiceLog', () => {
         console.log('it worked');
      });
  }
}
module.exports = MyPlugin;
```

Then
```js
const myPlugin = require('./myPlugin.js');
config.plugin('then').use(myPlugin);
```

<img width="424" alt="example" src="https://user-images.githubusercontent.com/17883920/43395367-6b558f38-9430-11e8-8f1a-1c449e99e9c3.PNG">

**Comment by yyx990803** [comment]
FYI I don't know how this error could possibly happen given these two places where `__expression` is actually used:

https://github.com/mozilla-neutrino/webpack-chain/blob/c31f4c6e8ed4dd7aec2826e96cec053f68f0d1f3/src/Plugin.js#L38-L39 - this can only happen if `this.get('plugin')` is undefined which means `plugin.use()` was never called.

https://github.com/mozilla-neutrino/webpack-chain/blob/9384537269d60bb80b3330cf44ddbdd9d528c454/src/Config.js#L84-L85 - this can never happen given that `typeof value === 'function'`.

The fact that object literal plugins do not work is unrelated to the `toString()` implementation and is fixed by #86.

**Comment by edmorley** [comment]
Ah true - thank you for #86 even though it was unrelated :-)
